plugins/network-elements/bigswitch/src/com/cloud/network/bigswitch/BigSwitchBcfApi.java

```diff
@@ -296,7 +296,7 @@ private void setHttpHeader(final HttpMethodBase m) {
         m.setRequestHeader(CONTENT_TYPE, CONTENT_JSON);
         m.setRequestHeader(ACCEPT, CONTENT_JSON);
         m.setRequestHeader(HTTP_HEADER_INSTANCE_ID, CLOUDSTACK_INSTANCE_ID + "-" + zoneId);
-        if (hash != "" ) {
+        if (! "".equals(hash)) {
```

Comment:
readability was my only concern plus it also has a not null check(though I dont know if hash can be null)